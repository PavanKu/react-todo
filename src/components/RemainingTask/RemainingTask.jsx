import React from "react";
import PropTypes from "prop-types";
import StyledText from "./RemainingTask.styled";

const DEFAULT_COUNT = 0;
export const NO_ITEM_LEFT = "Hurray! No item left";

function RemainingTask({ count = DEFAULT_COUNT }) {
  let text = NO_ITEM_LEFT;
  if (count && count > 0) {
    text = `${count} ${count > 1 ? "items" : "item"} left`;
  }

  return <StyledText>{text}</StyledText>;
}

RemainingTask.propTypes = {
  count: PropTypes.number,
};

export default RemainingTask;
