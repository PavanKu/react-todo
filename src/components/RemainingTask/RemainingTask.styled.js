import styled from "styled-components";

const StyledText = styled.div`
  font-size: 14px;
  color: #777;
  background: transparent;
`;

export default StyledText;
