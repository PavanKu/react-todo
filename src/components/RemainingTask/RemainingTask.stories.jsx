import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, number } from "@storybook/addon-knobs";

import RemainingTask from "./RemainingTask";

const ZeroTaskLeft = () => <RemainingTask />;

const OneTaskLeft = () => <RemainingTask count={1} />;

const MoreThanOneTaskLeft = () => <RemainingTask count={13} />;

const Playground = () => <RemainingTask count={number("count", 3)} />;

storiesOf("Remaing Item", module)
  .addDecorator(withKnobs)
  .add("Zero Task Left", ZeroTaskLeft)
  .add("One Task Left", OneTaskLeft)
  .add("Multiple Task Left", MoreThanOneTaskLeft)
  .add("Playground", Playground);
