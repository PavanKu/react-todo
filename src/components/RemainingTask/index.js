import RemainingTask from "./RemainingTask";

export { RemainingTask };
