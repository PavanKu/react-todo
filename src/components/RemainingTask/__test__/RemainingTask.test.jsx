import React from "react";
import { cleanup, render } from "@testing-library/react";

import RemainingTask, { NO_ITEM_LEFT } from "../RemainingTask";

describe("<RemainingTask />", () => {
  afterEach(cleanup);

  it("should render without crash", () => {
    render(<RemainingTask />);
  });

  it("should render with default props", () => {
    const { getByText } = render(<RemainingTask />);

    expect(getByText(NO_ITEM_LEFT)).toBeDefined();
  });

  it("should render with negative count", () => {
    const remainingTaskText = NO_ITEM_LEFT;
    const { getByText } = render(<RemainingTask count={-1} />);

    expect(getByText(remainingTaskText)).toBeDefined();
  });

  it("should render with singular count", () => {
    const remainingTaskText = `1 item left`;
    const { getByText } = render(<RemainingTask count={1} />);

    expect(getByText(remainingTaskText)).toBeDefined();
  });

  it("should render with plural count", () => {
    const remainingTaskText = "3 items left";
    const { getByText } = render(<RemainingTask count={3} />);

    expect(getByText(remainingTaskText)).toBeDefined();
  });

  it("should match 0 count snapshot", () => {
    const { container } = render(<RemainingTask count={0} />);
    expect(container).toMatchSnapshot();
  });

  it("should match singular count snapshot", () => {
    const { container } = render(<RemainingTask count={1} />);
    expect(container).toMatchSnapshot();
  });
  it("should match plural count snapshot", () => {
    const { container } = render(<RemainingTask count={3} />);
    expect(container).toMatchSnapshot();
  });
});
