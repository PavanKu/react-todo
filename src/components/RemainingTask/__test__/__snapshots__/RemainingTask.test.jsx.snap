// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemainingTask /> should match 0 count snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm gmZGCr"
  >
    Hurray! No item left
  </div>
</div>
`;

exports[`<RemainingTask /> should match plural count snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm gmZGCr"
  >
    3 items left
  </div>
</div>
`;

exports[`<RemainingTask /> should match singular count snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm gmZGCr"
  >
    1 item left
  </div>
</div>
`;
